src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java

```diff
@@ -0,0 +1,169 @@
+
+public class VoteShowValidationTest extends CreatedDefaultVoteRegistry {
+
+	Vote vote;
+	String voteId;
+	String adminKey;
+	
+	public void setUp() throws ReportedException {
+		super.setUp();
+		vote = voteManager.getVote(adminInfo.voteId);
+		voteId = adminInfo.voteId;
+		adminKey = adminInfo.adminKey;
+	}
+
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("validates inputs")
+	@Test
+	public void invalid_voteId_is_rejected() throws ReportedException {
+		String invalidvoteId = RandomUtils.createRandomKey();
+		assertThrows(
+			() -> voteManager.showVote(invalidvoteId, adminInfo.adminKey)
+		).assertMessageIs(String.format("illegal voteId: %s",invalidvoteId));
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("validates inputs")
+	@Test
+	public void invalid_adminKey_is_rejected() throws ReportedException {
+		String invalidAdminKey = RandomUtils.createRandomKey();
+		assertThrows(
+			() -> voteManager.showVote(adminInfo.voteId, invalidAdminKey)
+		).assertMessageIs(String.format("Illegal adminKey: %s",invalidAdminKey));
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void check_name() throws ReportedException {
+		JSONObject result = voteManager.showVote(voteId, adminKey);
+		
+		assertEquals(result.get("name"), vote.name);
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void check_canAddIn() throws ReportedException {
+		JSONObject result = voteManager.showVote(voteId, adminKey);
+		
+		assertEquals(result.get("canAddIn"), vote.canAddin);
+	}
+
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void check_creationTime() throws ReportedException {
+		JSONObject result = voteManager.showVote(voteId, adminKey);
+		
+		assertEquals(result.get("creationTime"), vote.creationTime);
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void check_choices() throws ReportedException {
+		JSONObject result = voteManager.showVote(voteId, adminKey);
+		JSONObject jsonObject = new JSONObject(vote.choices);
+
+		assertEquals(result.get("choices").toString(), jsonObject.toString());
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void check_canEndorse() throws ReportedException {
+		JSONObject result = voteManager.showVote(voteId, adminKey);
+		
+		assertEquals(result.get("canEndorse"), vote.canEndorse);
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void check_countedAssurances() throws ReportedException {
```

Comment:
see above about the names, and also some (although subminimal) concerns about serialisation here.